nemo_skills/evaluation/evaluator/mmau_pro.py

```diff
@@ -0,0 +1,169 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import re
+from typing import Any
+
+from tqdm import tqdm
+
+from nemo_skills.evaluation.evaluator.base import BaseEvaluatorConfig
+from nemo_skills.utils import get_logger_name, nested_dataclass
+
+LOG = logging.getLogger(get_logger_name(__file__))
+
+
+@nested_dataclass(kw_only=True)
+class MMAUProEvaluatorConfig(BaseEvaluatorConfig):
+    """Configuration for MMAU-Pro instruction following evaluation."""
+
+    pass
+
+
+def eval_mmau_pro(cfg):
+    """Evaluate MMAU-Pro instruction following questions using AIF (Audio Instruction Following) format.
+
+    This evaluator handles instruction following evaluation for MMAU-Pro benchmark.
+    Other question types are handled by different evaluation methods:
+    - Closed-form questions: Evaluated by nvembed_judge.py using NVEmbed similarity matching
+    - Open-ended questions: Evaluated by LLM judge (Qwen) using judge/speechlm prompt config
+    """
+    eval_config = MMAUProEvaluatorConfig(**cfg)
+
+    jsonl_file = eval_config.input_file
+    LOG.info(f"Evaluating instruction following questions in {jsonl_file}")
+
+    with open(jsonl_file, "rt", encoding="utf-8") as fin:
+        data = [json.loads(line) for line in fin]
+
+    LOG.info(f"Processing {len(data)} instruction following samples")
+
+    for idx, sample in enumerate(tqdm(data, desc="Evaluating samples")):
+        data[idx] = evaluate_instruction_following_sample(sample)
+
+    # Write all results at once
+    with open(jsonl_file, "wt", encoding="utf-8") as fout:
+        for sample in data:
+            fout.write(json.dumps(sample) + "\n")
+
+    LOG.info(f"Evaluation completed for {jsonl_file}")
+
+
+def evaluate_instruction_following_sample(sample: dict[str, Any]) -> dict[str, Any]:
+    """Evaluate a single instruction following sample."""
+    sample = sample.copy()
+    generation = sample.get("generation", "").strip()
+
+    if not generation:
+        LOG.info("Empty generation detected for instruction following question")
+        sample["is_correct"] = False
+        sample["error"] = "empty_generation"
+        return sample
+
+    success = evaluate_aif_constraints(
+        generation, sample.get("task_identifier", ""), sample.get("kwargs", {}) or {}, sample
```

Comment:
again check if things are indeed optional or not. E.g. task_identifier looks like it might be required, then let's not use .get